HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -0,0 +1,231 @@
+|-----------------------------------|---------------------------------------------------------------------------------------------------|
+| `HederaToken token`               | The hedera token.                                                                                 |
+| `FixedFee[] fixedFees`            | The fixed fee for the token.                                                                      |
+| `FractionalFee[] fractionalFees`  | The fractional fee for the token.                                                                 |
+| `RoyaltyFee[] royaltyFees`        | The royalty fee for the token.                                                                    |
+| `uint64 totalSupply`              | The number of tokens (fungible) or serials (non-fungible) of the token                            |
+| `bool deleted`                    | Specifies whether the token is deleted or not                                                     |
+| `bool defaultKycStatus`           | Specifies whether the token kyc was defaulted with KycNotApplicable (true) or Revoked (false)     |
+| `bool pauseStatus`                | Specifies whether the token is currently paused or not                                            |
+| `string ledgerId`                 | The ID of the network ledger                                                                      |
+
+Note that the tokenId and type of token (fungible or non-fungible) is not specified in the `TokenInfo` struct - it is implied 
+by the specific precompile function being called (see the functions below).
+
+#### FungibleTokenInfo
+`FungibleTokenInfo` struct defines the basic properties of a Fungible Hedera Token:
+
+| Field                 | Meaning                                               |
+|-----------------------|-------------------------------------------------------|
+| `TokenInfo tokenInfo` | The shared hedera token info                          |
+| `uint32 decimals`     | The number of decimal places a token is divisible by  |
+
+#### NonFungibleTokenInfo
+`NonFungibleTokenInfo` struct defines the basic properties of a Non Fungible Hedera Token:
+
+| Field                     | Meaning                                                                                       |
+|---------------------------|-----------------------------------------------------------------------------------------------|
+| `TokenInfo tokenInfo`     | The shared hedera token info                                                                  |
+| `int64 serialNumber`     | The serial number of the nft                                                                   |
+| `address ownerId`         | The account id specifying the owner of the non fungible token                                 |
+| `uint32 creationTime`     | The epoch second at which the token was created.                                              |
```

Comment:
It should be `int64`, since we keep the value as `long`.

Reply:
Fixed